frontends/web/src/routes/account/walletconnect/components/incoming-pairing/incoming-pairing.tsx

```diff
@@ -61,12 +61,17 @@ export const WCIncomingPairing = ({
     setPairingLoading(true);
     try {
       const { id, params } = currentProposal;
-      const { requiredNamespaces } = params;
-      const eipList = Object.values(requiredNamespaces);
+      const { requiredNamespaces, optionalNamespaces } = params;
+      const eipList = Object.keys(requiredNamespaces).length !== 0 ? Object.values(requiredNamespaces) : Object.values(optionalNamespaces);
+      if (!eipList) {
+        alertUser('Connecting app request did not define necessary namespace');
+        await handleRejectPairing();
+        setPairingLoading(false);
```

Comment:
I couldn't reproduce this alert probably as it fallsback to optionalNamespace?

but should it continue here or stop and `return;`?

Reply:
would probably need a custom dapp to reproduce, since ideally, it shouldn't happen that the dapp doesn't provide any namespace

I added the return just to be safe, no need to continue really